src/app/modules/item/pages/item-edit/item-edit.component.ts

```diff
@@ -108,13 +109,21 @@ export class ItemEditComponent implements OnDestroy {
         return this.createItemService
           .create(newChild)
           .pipe(map(res => ({ id: res, ...child })));
-      }),
-    ).pipe(
-      switchMap(children => {
-        // Saving the news children to not recreate them if there is an error.
-        this.itemChanges.children = children;
+      })
+    );
+  }
+
+  private updateItem(): Observable<void> {
+    return forkJoin({
+      children: this.createChildren().pipe(defaultIfEmpty())
+    }).pipe(
+      switchMap((res: {children?: ChildDataWithId[]}) => {
+        if (res.children) {
+          // Saving the news children to not recreate them if there is an error.
```

Comment:
// save the new children (their ids) to prevent recreating them in case of error

Reply:
Done